### DISCUSSION TITLE
How to properly use the Update row function in automations?

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2889] [Help]
I am having an automation write to an external second DB. I am struggling to get an automation to update the row in the second DB. how can I match the two Row id's? I see their is an extra section Row ID that appears when you select update row just not sure how to use it with an external DB.

**Comment by mjashanks** [comment]
I've successfully set this up in the very latest version of Budibase - `0.9.151`, due to be released over the next few days. It's only possible with this new release.

My setup
1. I have 2 internal tables, Contacts and Contacts2 - but it should work the same way with external SQL tables, or custom queries.
2. I setup an automation that gets triggered on `Update Row` of `Contacts`
3. I add a `Query Rows` step (this is new). I use this to find the `id` of the row I want in `Contacts2`
4. I use `Update Row` to update a column of `Contacts2`, using a value from `Contacts`.

Video below - sneak peek!

https://user-images.githubusercontent.com/3524181/136173758-1438c1fb-c68d-4650-982d-f3ca6a9ebbbc.mp4

**Reply by tobehn** [reply]
looks promising! Great stuff.

**Reply by KernOil** [reply]
@mjashanks This is great! I look forward to the new release. This should fix the issue I have been having with Attachments, I can use the internal DB so attachments and all other cool features work very easily. Then export the data I want "like a view" so I can still retail my reports built off the SQL DB

**Reply by KernOil** [reply]
@mjashanks in the new automation tab how do you activate the new automation? are they automatically on?

**Reply by mjashanks** [reply]
Ah yes, we need to get these documented. 

Automations only run when published. So, yes always on in published apps.

You can test them in dev, by using the test functionality.

**Reply by KernOil** [reply]
Gotch! ok so can only test with the button"Finish and test automation" or in the published app not in the  app_dev_ application is that correct?

**Reply by mjashanks** [reply]
Yep, that's correct.

**Reply by KernOil** [reply]
Thank You.

**Reply by KernOil** [reply]
How are you getting {{ step.1.rows.0._id }} as your Row ID in the above video?

**Reply by mjashanks** [reply]
I just know 😕 !

Really need to document these things well.

So, the `step.1.rows` is available through the binding options on the left. then `.0` means "choose the first item". Then `._id` is the `_id` field on that item.

Use the automation testing feature. Once a step has run, you will be able to see the Inputs and Outputs of each step - that's how you know what fields you can use.

**Reply by KernOil** [reply]
Gotcha. ok I will play around with the variable a bit. Thank you for explaining.
